wagtail/snippets/models.py

```diff
@@ -40,19 +41,38 @@ def get_edit_url(self, instance):
             )
 
 
-def register_snippet(model):
+def register_snippet(model, viewset=None):
     if model not in SNIPPET_MODELS:
+        from wagtail.snippets.views.chooser import SnippetChooserViewSet
+        from wagtail.snippets.views.snippets import SnippetViewSet
+
         model.get_usage = get_object_usage
         model.usage_url = get_snippet_usage_url
         model.get_admin_base_path = get_admin_base_path
         model.get_admin_url_namespace = get_admin_url_namespace
-        SNIPPET_MODELS.append(model)
-        SNIPPET_MODELS.sort(key=lambda x: x._meta.verbose_name)
 
-        url_finder_class = type(
-            "_SnippetAdminURLFinder", (SnippetAdminURLFinder,), {"model": model}
+        if viewset is None:
+            viewset = getattr(model, "admin_viewset", SnippetViewSet)
+        if isinstance(viewset, str):
+            viewset = import_string(viewset)
+
+        admin_viewset = viewset(
+            model.get_admin_url_namespace(),
+            model=model,
+            url_prefix=model.get_admin_base_path(),
+        )
+
+        chooser_viewset = SnippetChooserViewSet(
+            f"wagtailsnippetchoosers_{model._meta.app_label}_{model._meta.model_name}",
+            model=model,
+            url_prefix=f"snippets/choose/{model._meta.app_label}/{model._meta.model_name}",
         )
```

Comment:
As there are two viewsets here, should we name the parameter in `register_snippet` to something like `admin_viewset` instead of just `viewset`? In case we want to allow customising the `chooser_viewset` as well.